src/WebAppDIRAC/WebApp/handler/FileCatalogHandler.py

```diff
@@ -235,191 +248,85 @@ def web_getFilesData(self):
+                name = tmp[0]
+                logic = tmp[1]
 
-            name = tmp[0]
-            logic = tmp[1]
+                if logic not in self.__operands:
+                    gLogger.always("Operand '%s' is not supported " % logic)
+                    continue
 
-            if logic not in ["in", "nin", "=", "!=", ">=", "<=", ">", "<"]:
-                gLogger.always("Operand '%s' is not supported " % logic)
-                continue
+                if name in meta:
+                    # check existence of the 'name' section
+                    if name not in req["selection"]:
+                        req["selection"][name] = dict()
 
-            if name in meta:
-                # check existence of the 'name' section
-                if name not in req["selection"]:
-                    req["selection"][name] = dict()
+                    # check existence of the 'sign' section
+                    if logic not in req["selection"][name]:
+                        if tmp[2] == "v":
+                            req["selection"][name][logic] = ""
+                        elif tmp[2] == "s":
+                            req["selection"][name][logic] = []
 
-                # check existence of the 'sign' section
-                if logic not in req["selection"][name]:
                     if tmp[2] == "v":
-                        req["selection"][name][logic] = ""
+                        req["selection"][name][logic] = tmp[3]
                     elif tmp[2] == "s":
-                        req["selection"][name][logic] = []
-
-                if tmp[2] == "v":
-                    req["selection"][name][logic] = tmp[3]
-                elif tmp[2] == "s":
-                    req["selection"][name][logic] += tmp[3].split(":::")
-        if "path" in self.request.arguments:
-            req["path"] = self.get_argument("path")
-        gLogger.always("REQ: ", req)
-        return req
-
-    @asyncGen
-    def web_getMetadataFilesInFile(self):
-        self.set_header("Content-type", "text/plain")
-        self.set_header("Content-Disposition", 'attachment; filename="error.txt"')
-        req = self.__request_file()
-        gLogger.always(req)
-        gLogger.debug("submit: incoming request %s" % req)
-        result = yield self.threadTask(self.fc.findFilesByMetadata, req["selection"], req["path"])
+                        req["selection"][name][logic] += tmp[3].split(":::")
+            if path:
+                req["path"] = path
 
+        gLogger.debug("submit: incoming request %s" % req)
+        result = self.fc.findFilesByMetadata(req["selection"], req["path"])
         if not result["OK"]:
             gLogger.error("submit: %s" % result["Message"])
-            self.finish({"success": "false", "error": result["Message"]})
-            return
+            return {"success": "false", "error": result["Message"]}
 
-        result = result["Value"]
         retStrLines = []
+        if len(result["Value"]) > 0:
```

Comment:
if `result["Value"]` is a list (dict?) then the list comprehension also works with an empty `result["Value"]`?